File: result/translate/chatgpt/java2python/python/PROGRAM_BEST_FIT_ALGORITHM_MEMORY_MANAGEMENT.py
def f_gold(blockSize, m, processSize, n):
    allocation = [-1] * n
    for i in range(n):
        bestIdx = -1
        for j in range(m):
            if blockSize[j] >= processSize[i]:
                if bestIdx == -1:
                    bestIdx = j
                elif blockSize[bestIdx] > blockSize[j]:
                    bestIdx = j
        if bestIdx != -1:
            allocation[i] = bestIdx
            blockSize[bestIdx] -= processSize[i]
    
    print("\nProcess No.\tProcess Size\tBlock no.")
    for i in range(n):
        print("   " + str(i + 1) + "\t\t" + str(processSize[i]) + "\t\t", end="")
        if allocation[i] != -1:
            print(allocation[i] + 1)
        else:
            print("Not Allocated")

File: dataset/java2python/java/PROGRAM_BEST_FIT_ALGORITHM_MEMORY_MANAGEMENT.java
void f_gold ( int blockSize [ ] , int m , int processSize [ ] , int n ) {
  int allocation [ ] = new int [ n ] ;
  for ( int i = 0 ;
  i < allocation . length ;
  i ++ ) allocation [ i ] = - 1 ;
  for ( int i = 0 ;
  i < n ;
  i ++ ) {
    int bestIdx = - 1 ;
    for ( int j = 0 ;
    j < m ;
    j ++ ) {
      if ( blockSize [ j ] >= processSize [ i ] ) {
        if ( bestIdx == - 1 ) bestIdx = j ;
        else if ( blockSize [ bestIdx ] > blockSize [ j ] ) bestIdx = j ;
      }
    }
    if ( bestIdx != - 1 ) {
      allocation [ i ] = bestIdx ;
      blockSize [ bestIdx ] -= processSize [ i ] ;
    }
  }
  System . out . println ( "\nProcess No.\tProcess Size\tBlock no." ) ;
  for ( int i = 0 ;
  i < n ;
  i ++ ) {
    System . out . print ( "   " + ( i + 1 ) + "\t\t" + processSize [ i ] + "\t\t" ) ;
    if ( allocation [ i ] != - 1 ) System . out . print ( allocation [ i ] + 1 ) ;
    else System . out . print ( "Not Allocated" ) ;
    System . out . println ( ) ;
  }
}